thunder/core/symbol.py

```diff
@@ -321,9 +321,8 @@ def __call__(self, *args, **kwargs):
             # vjp transform (applied later).
             def tag_tensorproxy_output_as_detached(proxy):
                 if isinstance(proxy, TensorProxy):
-                    # We need to remove name from trace, otherwise replace will return a proxy with new name.
-                    trace.names.remove(proxy.name)
-                    return proxy.replace(tags=(ProxyTag.DETACHED_AUTOGRAD_GRAPH,))
+                    proxy.tags.add(ProxyTag.DETACHED_AUTOGRAD_GRAPH)
```

Comment:
This is a bit of a funny change, I am seeing the traces for autograd functions having two versions of the proxy, but have not managed to isolate the problem in the old autodiff setting. Apparently it re-executes things more so it does not hit the mismatch in any of the checked traces.

I am relatively confident though, that the trick here is problematic in that it introduces a collision - two distinct proxies with the same name.